### PR TITLE
Switch to node-fetch.

### LINKED ISSUE #242: Redirected URLs are not being followed
`Error: fetching the resource returned a 301 error code.`
This error pops out while clicking the hash button of any url.

### PR DESCRIPTION
**Pull request by XhmikosR**
~~This is a naive approach mostly a POC. Tests fail, but if you run this locally and try a URL which redirects, it will work.~~

Any help welcome.

This handles gzip, redirects, timeouts etc.

- [x] Add tests

Fixes #242.

### CODE REVIEW

File: lib/fetchResource.js [hunk 1]
```diff
@@ -36,6 +36,10 @@ const fetchResource = (resourceUrl, cb) => {
     let output;
     let data = [];
 
+    if (response.statusCode < 400 && response.headers.location) {
```
Comment (Johann-S) [inline lib/fetchResource.js]: redirection statusCode are between 300 and 308 so you can be strictier here

**Comment by Johann-S** [comment]
@XhmikosR #289 should do the trick 😉

**Comment by XhmikosR** [comment]
No need to close this @mozfreddyb. I still believe it's a bug and we need a solution either way.

**Comment by XhmikosR** [comment]
This works pretty good!

Only issue, maybe not an important one, is that we don't show anything while following redirects. But I think this could be tackled later if anyone wants to.

Huge thanks to @Johann-S :)

**Pull request merged**